src/fs/main.lua

```diff
@@ -135,11 +137,14 @@ do
     return { fd = fd, node = node }
   end
 
-  local function verify_fd(fd)
+  local function verify_fd(fd, dir)
     checkArg(1, fd, "table")
     if not (fd.fd and fd.node) then
       error("bad argument #1 (file descriptor expected)", 2)
     end
+    if (not not fd.dir) ~= (not not dir) then
+      error("bad argument #1 (cannot supply dirfd where fd is required, or vice versa)", 2)
+    end
```

Comment:
How do you even....

Reply:
please elaborate

Reply:
it converts both things to booleans to ensure neither side of the comparison is `nil` - should i add a comment clarifying that?

Reply:
> should i add a comment clarifying that?

Would be good.

Reply:
right, done